### PR TITLE
Add initial support for Mongo

### PR DESCRIPTION
**Pull request by ronichoudhury**
I'd like some code review for the express purpose of patching any severely broken bones in my "walking skeleton" version of the Mongo database connector.

@manthey, thanks for your help so far.

### CODE REVIEW

File: server/dbs/mongo.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    def connect(self):
+        self.conn = MongoClient(self.databaseUrl)
+
+    def disconnect(self):
+        self.conn.close()
+        self.conn = None
+
+    def performSelect(self, fields, queryProps={}, filters=[], client=None):
+        print "hiiii"
```
Comment (manthey) [inline server/dbs/mongo.py]: Debug code.

File: server/dbs/mongo.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+                    v = None
+            elif k == 'offset':
+                target = 'skip'
+            elif k in ['filter', 'limit', 'no_cursor_timeout', 'cursor_type', 'sort', 'allow_partial_results', 'oplog_replay', 'modifiers']:
+                target = k
+
+            if target is not None:
+                opts[target] = v
+
+        result['data'] = list(coll.find(**opts))
```
Comment (manthey) [inline server/dbs/mongo.py]: When returning data as a list, I was expecting a list-of-lists, with column information specified by the fields.  We should, ideally, either make the base class capable of handling both list-of-lists and list-of-dicts automatically, or we should transform this into the expected form.

The list-of-lists format uses vastly less bandwidth, which matters when querying hundreds of thousands of records.

**Comment by manthey** [comment]
Would it be useful to attempt to determine some of the fields in the collection?  In Postgres we can ask what column exist, but in Mongo we would ask for one or several records and assume that they are characteristic.

**Comment by ronichoudhury** [comment]
That would be useful. As per our conversation just now, I'll roll this feature into the list-of-lists computation and I'll file a new issue around pulling that out and putting it somewhere useful (maybe that `getFieldInfo()` method?).

**Pull request merged**